src/main/java/org/znerd/confluence/client/http/ConfluenceRestClient.java

```diff
@@ -426,6 +444,31 @@ public void deleteLabelFromPage(String contentId, String labelName) {
         sendRequest(deleteLabelFromPageRequest, (ignored) -> null);
     }
     
+    public ArrayList<ConfluenceLabel> getLabelsFromPage(String contentId) {
+
+        ArrayList<ConfluenceLabel> labels = new ArrayList<ConfluenceLabel>();   
+        HttpGet getLabelsByContentIdRequest = this.httpRequestFactory.getLabelsByContentIdRequest(contentId);
+
+        sendRequestAndFailIfNot20x(getLabelsByContentIdRequest, (response) -> {
+            JsonNode jsonNode = parseJsonResponse(response);
+    
+            int numberOfResults = jsonNode.get("size").asInt();
+            if (numberOfResults == 0) {
+                throw new NotFoundException();
+            }
+            if (numberOfResults > 0) {
+                for (Iterator<JsonNode> labelIterator = jsonNode.withArray("results").elements(); labelIterator.hasNext();) {
+                    ConfluenceLabel confluenceLabel = extractLabelFromJsonNode(labelIterator.next());
+                    labels.add(confluenceLabel);
+                }
+            }
+            
+            return labels;
+        });
+        
+        return labels;
```

Comment:
Why is `labels` returned both from within the lamda function, and from the method? I don't understand this part of the code, pardon my ignorance.